src/app/pages/subjects/subjects.js

```diff
@@ -0,0 +1,167 @@
+                books: books.filter((aBook) => aBook.categories.indexOf(category) >= 0)
+                    .map((book) => ({
+                        title: book.title,
+                        coverUrl: book.coverUrl
+                    }))
+            }
+        ));
+
+    return {
+        filterButtons,
+        booksByCategory
+    };
+};
+
+function stateData() {
+    let selectedFilter = 'View All',
+        selectedBook = null;
+
+    return {
+        selectedFilter,
+        selectedBook
+    };
+}
+
+@props({
+    template: template,
+    regions: {
+        filterButtons: '.filter-buttons',
+        bookViewer: '.book-viewer'
+    }
+})
+export default class Subjects extends BaseView {
+
+    constructor() {
+        super();
+        this.model = new BaseModel({viewIsReady: false});
+
+        let updateBookIndexInfo = () => {
+            let setHtmlToMember = (nodeName, memberName) => {
+                this.el.querySelector(`[data-manager="${nodeName}"]`)
+                .innerHTML = this.model.get('bookIndex')[memberName];
+            };
+
+            if (this.model.get('viewIsReady') &&
+                this.model.has('bookIndex')) {
+                setHtmlToMember('page-description', 'page_description');
+                setHtmlToMember('ds1-head', 'dev_standard_1_heading');
+                setHtmlToMember('ds1-body', 'dev_standard_1_description');
+                setHtmlToMember('ds2-head', 'dev_standard_2_heading');
+                setHtmlToMember('ds2-body', 'dev_standard_2_description');
+                setHtmlToMember('ds3-head', 'dev_standard_3_heading');
+                setHtmlToMember('ds3-body', 'dev_standard_3_description');
+            }
+        };
+
+        this.model.on('change:bookIndex change:viewIsReady', updateBookIndexInfo);
+
+        this.model.on('change:filterButtons change:viewIsReady', () => {
+            if (this.model.get('viewIsReady') &&
+                this.model.has('filterButtons')) {
+                let buttons = this.model.get('filterButtons'),
+                    region = this.regions.filterButtons;
+
+                for (let button of buttons) {
+                    region.append(new FilterButton(button, this.model));
+                }
+            }
+        });
+
+        this.model.on('change:booksByCategory change:viewIsReady', () => {
+            if (this.model.get('viewIsReady') &&
+                this.model.has('booksByCategory')) {
+                let categories = this.model.get('booksByCategory'),
+                    region = this.regions.bookViewer;
+
+                for (let category of categories) {
+                    region.append(new CategorySection(category, this.model));
+                }
+            }
+        });
+
+        this.templateHelpers = {};
+        setTimeout(() => {
+            this.model.set('bookIndex', jsonData());
+        }, 800);
+
+        setTimeout(() => {
+            this.model.set(dummyData());
+        }, 300);
```

Comment:
I think simulating data loading on a delay is fine for testing, but I'm not sure it should find its way into master (unless it's part of an actual test).